pandas/src/algos_common_helper.pxi.in

```diff
@@ -83,8 +83,8 @@ def pad_{{name}}(ndarray[{{c_type}}] old, ndarray[{{c_type}}] new,
     if limit is None:
         lim = nright
     else:
-        if limit < 0:
-            raise ValueError('Limit must be non-negative')
+        if limit < 1:
```

Comment:
alternatively you can simply remove the cython check entirely (see if anything breaks), it might not even be hitting it (and instead catching it at a higher level, IOW your new checks in missing.py), not sure.

```
if limit is None:
    .....
else:
    lim = int(limit)
```
which is will fail hard if its not an integer, and will work (incorrrectly) if its a float, so have to be careful here.